dojo/problem/views.py

```diff
@@ -116,14 +144,16 @@ def get_problems(self, request: HttpRequest):
 class ListClosedProblems(ListProblems):
     filter_name = "Closed"
 
-    def get_problems(self, request: HttpRequest):
+    def get_problems(self, request: HttpRequest, products=None):
+        user_fids, active_fids = self.get_findings(products)
         list_problem = []
-        active_findings = set(
-            Finding.objects.filter(id__in=[fid for p in self.problems_map.values() for fid in p.finding_ids], active=True)
-            .values_list("id", flat=True),
-        )
         for _, problem in self.problems_map.items():
-            if not any(finding_id in active_findings for finding_id in problem.finding_ids):
+            problem.finding_ids = set(problem.finding_ids) & user_fids
+            if not problem.finding_ids:
+                continue
+            if not (problem.finding_ids & active_fids):
```

Comment:
Is this `not` correct? Seems like we want to enter the `if` if the set is has elements.